src/mongocxx/test/instance.cpp

```diff
@@ -0,0 +1,69 @@
+// Copyright 2015 MongoDB Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "catch.hpp"
+#include "helpers.hpp"
+
+#include <vector>
+
+#include <bsoncxx/stdx/make_unique.hpp>
+
+#include <mongocxx/instance.hpp>
+#include <mongocxx/logger.hpp>
+#include <mongocxx/private/libmongoc.hpp>
+#include <mongocxx/stdx.hpp>
+
+namespace {
+using namespace mongocxx;
+
+class test_log_handler : public logger {
+   public:
+    using event = std::tuple<log_level, std::string, std::string>;
+
+    test_log_handler(std::vector<event>* events) : _events(events) {
+    }
+
+    std::vector<event>* _events;
+
+    void operator()(log_level level, stdx::string_view domain,
+                    stdx::string_view message) noexcept final {
+        try {
+            _events->emplace_back(level, std::string(domain), std::string(message));
+        } catch (...) {
```

Comment:
The function is noexcept, so if it throws it will automatically call terminate. I think you can drop the try/catch